### ISSUE TITLE
Hide unnecessary buttons in result view

### ISSUE THREAD

**Issue by lgeiger** [nteract/hydrogen#796]
Depending on the output not every button is useful. To make the appearance cleaner we should hide unnecessary buttons:
![bildschirmfoto 2017-05-21 um 00 36 53](https://cloud.githubusercontent.com/assets/13285808/26279923/6cd624f0-3dc2-11e7-893c-f2ede159fd6d.png)

- [x] The [copy](https://github.com/nteract/hydrogen/blob/master/lib/components/result-view/result-view.js#L116) and [open in editor](https://github.com/nteract/hydrogen/blob/master/lib/components/result-view/result-view.js#L121)  should only be shown if there's actually [text to copy](https://github.com/nteract/hydrogen/blob/master/lib/components/result-view/result-view.js#L23).
- [x] The [expand button](https://github.com/nteract/hydrogen/blob/master/lib/components/result-view/result-view.js#L125) should only be shown if the result doesn't fit in the output box.
Pseudo code (not tested):
```js
import { DEFAULT_SCROLL_HEIGHT } from "@nteract/display-area"

...

{(this.el.scrollHeight > DEFAULT_SCROLL_HEIGHT`)
  ? <div
      className={`icon icon-${this.expanded.get() ? "fold" : "unfold"}`}
      onClick={this.toggleExpand}
    />
  : null}
...
```

**Comment by nikitakit** [comment]
Note that the result number (execution count) is currently shown on-hover over the copy button, so if the copy button is hidden there may need to be some alternative way of surfacing that information.

**Comment by lgeiger** [comment]
> Note that the result number (execution count) is currently shown on-hover over the copy button, so if the copy button is hidden there may need to be some alternative way of surfacing that information.

Good point. What about moving the tooltip to the close button?

**Comment by nikitakit** [comment]
Not sure about that. The information provided may be only tangentially related to the copy action, but it's even less related to the close action. Plus, if your fingers slip and you end up clicking then the bubble is gone -- and the result number is precisely what you need to know to bring it back.

This might be an OK compromise in the interim, but I really think there should be a better way of providing this information.

**Comment by lgeiger** [comment]
@nikitakit I agree.

I think with the new unified result view (and the removal of the rich display) we can find a place to permanently show the execution count in the result view. I'm very open to ideas!

By the way how do you bring back the result based on the execution count?

**Comment by nikitakit** [comment]
@lgeiger 

A agree that it would be nice if we could sneak in the result number somewhere. I think it wouldn't look too bad to put it right under the close button (basically an element like a button, but which displays the number and isn't clickable.) That said I don't think my CSS is good enough to code that myself.

In terms of bringing results back, you can't return the print statements, but you can refer to the output itself using variables like `Out[1]` or `_1`, at least in python.

**Comment by lgeiger** [comment]
Good idea!
We could also experiment with putting it in the gutter right next to the result view.

> In terms of bringing results back, you can't return the print statements, but you can refer to the output itself using variables like Out[1] or _1, at least in python.

Woo didn't know that! That's pretty handy.

**Comment by BenRussert** [comment]
> The copy and open in editor should only be shown if there's actually text to copy

If we were to make this change where would the execution count go? I like using the close button tooltip if you wanted to revisit that idea.

If that doesn't work for everybody, I experimented with this some just thinking about alternatives. Below is displaying execution count right in the toolbar. This would be very simple to do (could probably  move it and add some styling to make it look nicer):

![absolutely-insane-crazy-idea](https://user-images.githubusercontent.com/10860657/29243820-28f4fdc2-7f6e-11e7-9018-c777b7cb2743.png)

**Comment by lgeiger** [comment]
> I like using the close button tooltip if you wanted to revisit that idea.

👍

**Comment by lgeiger** [comment]
Closed in #943 and #859
